Library/Homebrew/brew.rb

```diff
@@ -37,7 +37,7 @@
   homebrew_path = PATH.new(ENV["HOMEBREW_PATH"])
 
   # Add SCM wrappers.
-  path.append(HOMEBREW_SHIMS_PATH/"scm")
+  path.prepend(HOMEBREW_SHIMS_PATH/"scm")
   homebrew_path.append(HOMEBREW_SHIMS_PATH/"scm")
```

Comment:
Should this also be changed to `prepend`?

Reply:
Or to put it another way, is it likely that an external command would want to use the `git` specified by `HOMEBREW_GIT` (e.g. if set with `HOMEBREW_GIT_PATH`) rather than whichever is first in the user's `PATH`? I'd say yes.